drivers/timer/rcar_cmt_timer.c

```diff
@@ -150,3 +150,8 @@ uint32_t sys_clock_cycle_get_32(void)
 {
 	return sys_read32(TIMER_BASE_ADDR + CMCNT1_OFFSET);
 }
+
+uint64_t sys_clock_cycle_get_64(void)
+{
+	return 0;
```

Comment:
@julien-massot - would be great to get your input here. It seems there is no 64-bit timer register on this platform.

For software emulation, I would like to increment a 32-bit "top half" each time a 32-bit cycle counter overflows (assuming a 1:1 ratio), shift and then OR it with the regular 32-bit value.

In this driver, where is the best location to increment the 32-bit "top half"?

Reply:
Hi @cfriedt, what I can do is to connect the overflow interrupt and increments a 32 bit top half.
But one rule about the sys_clock_cycle_get_32 is that it should work even with irq disabled, to allow k_busy_wait
to work with irq locked or within high priority ISR.  

So in case we increments the top half in the overflow interrupts, sys_clock_cycle_get_32 will be reliable
with irq locked and sys_clock_cycle_get_64 will not.

Reply:
Interesting - @stephanosio also suggested another solution for when overflow IRQs are not available is to use the 64-bit system tick counter + some arithmetic for the upper 32-bits.